### ISSUE TITLE
chezmoi: .local/man: duplicate externals

### ISSUE THREAD

**Issue by stablestud** [twpayne/chezmoi#3232]
## Describe the bug

When multiple external archive entries are setup to output to the same directory, chezmoi fails with the error message: 
`chezmoi: .local/man: duplicate externals`

## To reproduce

Here for example I extract the manpages from both archives.
These must be placed into the same directory

```toml
[".local/man"]
# source: https://github.com/junegunn/fzf
type = "archive"
include = ["fzf-0.42.0/man/**"]
url = "https://github.com/junegunn/fzf/archive/refs/tags/0.42.0.tar.gz"
stripComponents = 2

[".local/man"]                                          
# source: https://github.com/ajeetdsouza/zoxide 
type = "archive"                                
include = ["man/**"]
url = "https://github.com/ajeetdsouza/zoxide/releases/download/v0.9.2/zoxide-0.9.2-x86_64-unknown-linux-musl.tar.gz"
stripComponents = 1
```

## Expected behavior

Files are added from both archives.

```console
.local/man
└── man1
    ├── fzf.1
    ├── fzf-tmux.1
    ├── zoxide.1
    ├── zoxide-add.1
    ├── zoxide-import.1
    ├── zoxide-init.1
    ├── zoxide-query.1
    └── zoxide-remove.1
```

## Output of `chezmoi doctor`

<details>

```console
$ chezmoi doctor
RESULT   CHECK                       MESSAGE
ok       version                     v2.39.1, commit fe10a696fb9aad76776bce62f9fa9a4363bac62a, built at 2023-09-06T12:50:07Z, built by goreleaser
ok       latest-version              v2.39.1
ok       os-arch                     linux/amd64 (Ubuntu 22.04.3 LTS (Jammy Jellyfish))
ok       uname                       Linux ber-040 5.15.0-83-generic #92-Ubuntu SMP Mon Aug 14 09:30:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
ok       go-version                  go1.21.0 (gc)
ok       executable                  /snap/chezmoi/749/chezmoi
ok       upgrade-method              snap-refresh
ok       config-file                 ~/.config/chezmoi/chezmoi.toml, last modified 2023-09-13T10:12:50+02:00
ok       source-dir                  ~/.local/share/chezmoi is a git working tree (clean)
ok       suspicious-entries          no suspicious entries
ok       working-tree                ~/.local/share/chezmoi is a git working tree (clean)
ok       dest-dir                    ~ is a directory
ok       umask                       002
ok       cd-command                  found /usr/bin/fish
ok       cd-args                     /usr/bin/fish
info     diff-command                not set
ok       edit-command                found /usr/bin/vi
ok       edit-args                   /usr/bin/vi
ok       git-command                 found /usr/bin/git, version 2.34.1
ok       merge-command               found /usr/bin/vimdiff
ok       shell-command               found /usr/bin/fish
ok       shell-args                  /usr/bin/fish
info     age-command                 age not found in $PATH
ok       gpg-command                 found /usr/bin/gpg, version 2.2.27
info     pinentry-command            not set
info     1password-command           op not found in $PATH
info     bitwarden-command           bw not found in $PATH
info     bitwarden-secrets-command   bws not found in $PATH
info     dashlane-command            dcli not found in $PATH
info     doppler-command             doppler not found in $PATH
info     gopass-command              gopass not found in $PATH
ok       keepassxc-command           found /usr/bin/keepassxc-cli, version 2.6.6
info     keepassxc-db                not set
info     keeper-command              keeper not found in $PATH
info     lastpass-command            lpass not found in $PATH
info     pass-command                pass not found in $PATH
info     passhole-command            ph not found in $PATH
info     rbw-command                 rbw not found in $PATH
info     vault-command               vault not found in $PATH
info     vlt-command                 vlt not found in $PATH
info     secret-command              not set
```

</details>

**Comment by twpayne** [comment]
Thanks for reporting this. I'd argue that this is a feature request to allow overlapping, non-conflicting externals, rather than a bug.

Note that the TOML file that you post in the "To reproduce" section is not valid as it contains duplicate keys (`".local/man"`) which are not allowed, so these must be two separate `.chezmoiexternal` files.

However, the use case makes sense and I've implemented it in #3234.

**Comment by stablestud** [comment]
Thank you for implementing :+1: 
Looking forward to the release :smiley:
